src/ComposerRequireChecker/Cli/ResultsWriter/JsonFile.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ComposerRequireChecker\Cli\ResultsWriter;
+
+use DateTimeImmutable;
+use RuntimeException;
+
+use function dirname;
+use function file_put_contents;
+use function function_exists;
+use function is_dir;
+use function json_encode;
+use function mkdir;
+use function sprintf;
+
+final class JsonFile implements ResultsWriter
+{
+    private string $filePathname;
+    private string $applicationVersion;
+
+    public function __construct(string $filePathname, string $applicationVersion)
+    {
+        if (! function_exists('json_encode')) {
+            throw new RuntimeException('Missing extension: json');
+        }
+
+        $this->filePathname       = $filePathname;
+        $this->applicationVersion = $applicationVersion;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function write(array $unknownSymbols): void
+    {
+        $filePath = dirname($this->filePathname);
+        if (! mkdir($filePath, 0777, true) && ! is_dir($filePath)) {
```

Comment:
`0777` seems... aggressive. I'd say that the tool should not create any directories: the file must already be provided. 

Alternatively: remove all the file writing, and write the JSON to STDOUT, for redirection via bash built-ins (IMO simpler solution)

Reply:
`0777` is the param default. But I agree, directory should exist already. I’ll change it to fail with exception otherwise.